src/services/common/measurement.utils.ts

```diff
@@ -1,73 +1,73 @@
+import { formatLength } from '@/directives/format-length.directive'
 import { Coordinate } from 'ol/coordinate'
-import { LineString, Polygon, Point, Geometry } from 'ol/geom'
-import { Projection, transform } from 'ol/proj'
-import { getDistance as haversineDistance, getArea } from 'ol/sphere'
+import { LineString, Polygon, Point, Geometry, Circle } from 'ol/geom'
+import {
+  getPointResolution,
+  METERS_PER_UNIT,
+  Projection,
+  transform,
+} from 'ol/proj'
+import {
+  getDistance as haversineDistance,
+  getArea as getOlArea,
+} from 'ol/sphere'
+import { Map } from 'ol'
 
-const getFormattedLength = function (
-  geom: Geometry,
-  projection: Projection,
-  precision?: number
-): string {
+const getLength = function (geom: Geometry, projection: Projection): number {
   let length = 0
-  let coordinates
+  let coordinates: Coordinate[] = []
   if (geom.getType() === 'Polygon') {
     coordinates = (geom as Polygon).getCoordinates()[0] as Coordinate[]
   } else if (geom.getType() === 'LineString') {
     coordinates = (geom as LineString).getCoordinates() as Coordinate[]
-  } else {
-    return ''
   }
   for (let i = 0, ii = coordinates.length - 1; i < ii; ++i) {
     const c1 = transform(coordinates[i], projection, 'EPSG:4326')
     const c2 = transform(coordinates[i + 1], projection, 'EPSG:4326')
```

Comment:
Not very important and not related to your work, but there are some CONSTs for projections, we should use them. (this is just a side note, you can keep it as is if you don't have time for this modification ^^)